src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -1368,6 +1368,41 @@ void TorrentHandleImpl::renameFile(const int index, const QString &name)
     m_nativeHandle.rename_file(lt::file_index_t {index}, Utils::Fs::toNativePath(name).toStdString());
 }
 
+void TorrentHandleImpl::renameFolder(const QString &oldPath, const QString &newPath)
+{
+    if (!isValid()) return;
+    const QString sanatizedOldPath = oldPath + (oldPath.endsWith("/") ? "" : "/");
+    const QString sanatizedNewPath = newPath + (newPath.endsWith("/") ? "" : "/");
+
+    // Check for overwriting
+#if defined(Q_OS_WIN)
+    const Qt::CaseSensitivity caseSensitivity = Qt::CaseInsensitive;
+#else
+    const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive;
+#endif
+    for (int i = 0; i < filesCount(); ++i) {
+        const QString currentPath = filePath(i);
+
+        if (currentPath.startsWith(sanatizedOldPath))
+            continue;
+
+        if (currentPath.startsWith(sanatizedNewPath, caseSensitivity)) {
+            LogMsg(tr("The folder could not be renamed.").arg(name(), "This name is already in use."), Log::CRITICAL);
```

Comment:
This code is incorrect.

Reply:
Are you referring to the code logic or the `LogMsg`? I wasn't sure when to use `LogMsg` and when to use `qDebug`. I could use some direction here.

Reply:
@glassez